### ISSUE TITLE
Botón deshacer

### ISSUE THREAD

**Issue by nimonro** [neslerel/BetaTesters_QGRed#55]
Supongo que no será sencillo, pero el otro día estaba editando un modelo, se me fue el dedo y perdí varios cambios. 

Sería interesante poder tener un botón de deshacer los últimos cambios.

Ligado con ese fallo además creó que se corrompió el tema de Valves y dejó de funcionar la aplicación. El fallo ocurrió cuando estaba cambiando los nodos inicial y final de una válvula, me equivoque al seleccionar uno de los nodos en el desplegable y la válvula se fue a unir a otra válvula contigua. A partir de ahí ya falló y no pude seguir trabajando. 

Lo pude solucionar exportando el modelo a inp y volviéndolo a importar en una base de datos nueva. 

Gracias.

**Comment by neslerel** [comment]
Sí, es algo que nos gustaría tener, pero como dices es algo complicado... Tengo una idea de cómo hacerlo, pero no es que sea muy eficiente, así que tenemos que darle una vuelta.

Respecto al error, es verdad que al permitir cambiar nudos iniciales y finales deberíamos hacer comprobaciones y no permitir ciertas cosas que rompan el modelo o que no tengan sentido.
Lo apunto para solucionar.

Gracias por los comentarios!
